frontend/src/views/admin/assignments/offer-actions.js

```diff
@@ -120,53 +171,53 @@ function OfferActionButtons(props) {
 
     return (
         <React.Fragment>
-            <ActionButton
-                icon={<FaUserPlus />}
-                onClick={createOffers}
-                disabled={!actionPermitted.canCreate}
-            >
-                Create Offer
-            </ActionButton>
-            <ActionButton
-                icon={<FaUserTimes />}
-                onClick={confirmOfferWithdraw}
-                disabled={!actionPermitted.canWithdraw}
-            >
-                Withdraw Offer
-            </ActionButton>
-            <ActionButton
-                icon={<FaEnvelope />}
-                onClick={emailOffers}
-                disabled={!actionPermitted.canEmail}
-            >
-                Email Offer
-            </ActionButton>
-            <ActionButton
-                icon={<FaUserClock />}
-                onClick={nagOffers}
-                disabled={!actionPermitted.canNag}
-            >
-                Nag Offer
-            </ActionButton>
-            <ActionButton
-                icon={<FaCheck />}
-                onClick={acceptOffers}
-                disabled={!actionPermitted.canAccept}
-            >
-                Set as Accepted
-            </ActionButton>
-            <ActionButton
-                icon={<FaBan />}
-                onClick={rejectOffers}
-                disabled={!actionPermitted.canReject}
-            >
-                Set as Rejected
-            </ActionButton>
-            <MultiWithdrawOfferConfirmation
-                data={selectedAssignments}
-                visible={showWithdrawConfirmation}
-                setVisible={setShowWithdrawConfirmation}
-                withdrawOffers={withdrawOffers}
+            <CreateOfferButtonWithDialog
+                actionButtonEnable={actionPermitted.canCreate}
+                actionButtonOnClick={confirmOfferCreate}
+                selectedAssignments={selectedAssignments}
+                dialogVisible={showCreateConfirmation}
+                dialogSetVisible={setShowCreateConfirmation}
+                actionCallback={createOffers}
```

Comment:
Why is there a `callback` and an `onClick`? (Also, don't invent a new name when `onClick` will do). Create a closure that performs the correct action and pass it as the callback. No need for an `onClick`. The `CreateOfferButtonWithDialog` will handle showing the dialog if needed.